### ISSUE TITLE
Single letter domain email validation

### ISSUE THREAD

**Issue by artkrv** [FluentValidation/FluentValidation#2079]
### FluentValidation version

11.5.1

### ASP.NET version

.net 7

### Summary

I have a rule in my request for email property:
`RuleFor(x => x.Email).EmailAddress();
`
If user passes email@email.c validator does not detect it as an incorrect value.
(resultBad4 in tests).

### Steps to Reproduce

```
public class RegistrationRequest
{
     public string Email { get; set; }
 }

 public class RegistrationRequestValidator : AbstractValidator<RegistrationRequest>
 {
     public RegistrationRequestValidator()
     {
         RuleFor(x => x.Email).EmailAddress();
     }
 }


[Fact]
 public void EmailFormatValidator()
 {
     const string property = "Email";
     var validator = new RegistrationRequestValidator();

     var resultBad1 = validator.Validate(new RegistrationRequest(){Email = "email"});
     var resultBad2 = validator.Validate(new RegistrationRequest(){Email = "email.bad"});
     var resultBad3 = validator.Validate(new RegistrationRequest(){Email = "email@bad@.bad"});
     var resultBad4 = validator.Validate(new RegistrationRequest(){Email = "email@bad.b"});
     var resultOk1 = validator.Validate(new RegistrationRequest(){Email = "email@ok"});
     var resultOk2 = validator.Validate(new RegistrationRequest(){Email = "email@ok.ok"});

     Assert.Contains(resultBad1.Errors, x => x.PropertyName == property);
     Assert.Contains(resultBad2.Errors, x => x.PropertyName == property);
     Assert.Contains(resultBad3.Errors, x => x.PropertyName == property);
     Assert.Contains(resultBad4.Errors, x => x.PropertyName == property);
     Assert.DoesNotContain(resultOk1.Errors, x => x.PropertyName == property);
     Assert.DoesNotContain(resultOk2.Errors, x => x.PropertyName == property);
}

```

**Comment by JeremySkinner** [comment]
Please see https://github.com/FluentValidation/FluentValidation/issues/1895
